language/tools/move-bytecode-utils/src/lib.rs

```diff
@@ -0,0 +1,97 @@
+// Copyright (c) The Diem Core Contributors
+// SPDX-License-Identifier: Apache-2.0
+
+pub mod dependency_graph;
+
+use crate::dependency_graph::DependencyGraph;
+use move_binary_format::{access::ModuleAccess, file_format::CompiledModule};
+use move_core_types::language_storage::ModuleId;
+
+use anyhow::{anyhow, Result};
+use std::collections::BTreeMap;
+
+/// Set of Move modules indexed by module Id
+#[derive(Debug, PartialEq, Eq, Clone)]
+pub struct Modules<'a>(BTreeMap<ModuleId, &'a CompiledModule>);
+
+impl<'a> Modules<'a> {
+    /// Construct a set of modules from a slice `modules`.
+    /// Panics if `modules` contains duplicates
+    pub fn new(modules: impl IntoIterator<Item = &'a CompiledModule>) -> Self {
+        let mut map = BTreeMap::new();
+        for m in modules {
+            assert!(
+                map.insert(m.self_id(), m).is_none(),
+                "Duplicate module found"
+            );
+        }
+        Modules(map)
+    }
+
+    /// Return all modules in this set
+    pub fn iter_modules(&self) -> Vec<&CompiledModule> {
+        self.0.values().copied().collect()
+    }
+
+    /// Compute a dependency graph for `self`
+    pub fn compute_dependency_graph(&self) -> DependencyGraph {
+        DependencyGraph::new(self.0.values().copied())
+    }
+
+    /// Return the backing map of `self`
+    pub fn get_map(&self) -> &BTreeMap<ModuleId, &CompiledModule> {
+        &self.0
+    }
+
+    /// Return the bytecode for the module bound to `module_id`
+    pub fn get_module(&self, module_id: &ModuleId) -> Option<&CompiledModule> {
+        self.0.get(module_id).copied()
+    }
+
```

Comment:
I'm not sure how much benefit this function provides since a caller can just do `get_module_res(...).ok()` instead. What would you think of removing this function and just keeping the one that returns a `Result`?